packages/shared/src/components/RankProgress.tsx

```diff
@@ -109,7 +169,9 @@ export function RankProgress({
         )
       : null;
 
-    const firstBadgeAnimation = badgeRef.current.animate(
+    const firstBadgeAnimation = (
+      badgeRef.current ? badgeRef.current : plusRef.current
```

Comment:
Would probably be better off declaring this in a variable to only have a single point of change for maintainability. What do you think?

Like:
`const animatedRef = badgeRef.current || plusRef.current`

Reply:
Great idea, I'll refactor it